mypyc/lib-rt/dict_ops.c

```diff
@@ -80,6 +80,31 @@ PyObject *CPyDict_SetDefaultWithNone(PyObject *dict, PyObject *key) {
     return CPyDict_SetDefault(dict, key, Py_None);
 }
 
+PyObject *CPyDict_SetDefaultWithEmptyDatatype(PyObject *dict, PyObject *key,
+                                              int data_type) {
+    PyObject *res = CPyDict_GetItem(dict, key);
+    if (!res) {
+        PyErr_Clear();
```

Comment:
Do you have this because the function may return `NULL` without raising an exception in case `data_type` is unsupported? It would be cleaner to instead raise `RuntimeError` or similar instead of just returning `NULL`. At least it would be better to have this just before `return NULL;` to make it clearer why we are doing this.